snapcraft/storeapi/__init__.py

```diff
@@ -638,6 +658,64 @@ def push_metadata(self, snap_id, snap_name, metadata, force):
         if not response.ok:
             raise errors.StoreMetadataError(snap_name, response, metadata)
 
+    def push_binary_metadata(self, snap_id, snap_name, metadata, force):
+        """Push the binary metadata to SCA."""
+        url = 'snaps/' + snap_id + '/binary-metadata'
+        headers = {
+            'Authorization': _macaroon_auth(self.conf),
+            'Accept': 'application/json',
+        }
+
+        # get current binary metadata information
+        response = self.request('GET', url, headers=headers)
+        if not response.ok:
+            raise errors.StoreMetadataError(snap_name, response, metadata)
+
+        binary_metadata = response.json()
+        # current icons and screenshots
+        icons = [media for media in binary_metadata
+                 if media.get('type') == 'icon']
+        screenshots = [media for media in binary_metadata
+                       if media.get('type') == 'screenshot']
+
+        files = {}
+        # only icon support atm
+        icon = metadata.get('icon')
```

Comment:
this is going to end up being a rather big function, how about a private module inside `storeapi` to handle this with some OO? Nothing complex

what do you think?

Reply:
Let me take a look, will try to refactor it.